packages/core/src/routes/application.ts

```diff
@@ -79,17 +87,38 @@ export default function applicationRoutes<T extends AuthedRouter>(
     '/applications/:id',
     koaGuard({
       params: object({ id: string().min(1) }),
-      body: Applications.createGuard.omit({ id: true, createdAt: true }).deepPartial(),
+      body: Applications.createGuard
+        .omit({ id: true, createdAt: true })
+        .deepPartial()
+        .merge(
+          z.object({
+            isAdmin: z.boolean().optional(),
+          })
+        ),
     }),
     async (ctx, next) => {
       const {
         params: { id },
         body,
       } = ctx.guard;
 
-      ctx.body = await updateApplicationById(id, {
-        ...body,
-      });
+      const { isAdmin, ...rest } = body;
+
+      // FIXME @sijie temp solution to set admin access to machine to machine app
+      if (isAdmin !== undefined) {
+        const applicationsRoles = await findApplicationsRolesByApplicationId(id);
+        const originalIsAdmin = applicationsRoles.some(
+          ({ roleId }) => roleId === adminConsoleAdminRoleId
+        );
+
+        if (isAdmin && !originalIsAdmin) {
+          await insertApplicationsRoles([{ applicationId: id, roleId: adminConsoleAdminRoleId }]);
+        } else if (!isAdmin && originalIsAdmin) {
+          await deleteApplicationsRoles(id, adminConsoleAdminRoleId);
+        }
```

Comment:
nit:  not sure about this. But, since we have to check the `originalIsAdmin` from DB anyway, would it be ok to directly insert/delete a role without checking its existence?  Just catch the error i.e.

Reply:
Both are OK, I'll keep the old logic.